### PR TITLE
Add is_equal method for readers

### PR DESCRIPTION
**Pull request by guilhermebodin**
The question here is if we simply return false or return an AssertionError.
Either way we should display a message warning about what is not equal.

### CODE REVIEW

File: src/time_series_utils.jl [hunk 1]
```diff
@@ -187,4 +187,38 @@ function file_to_array_and_header(
...
+    
+    # Assert dimensions
+    @assert max_stages(ior1) == max_stages(ior2) "erro"
+    @assert max_scenarios(ior1) == max_scenarios(ior2) "erro"
+    @assert max_blocks(ior1) == max_blocks(ior2) "erro"
+
+    # Assert the agents are the same in both files
+    @assert agent_names(ior1) == agent_names(ior2)
+
+    for estagio = 1:max_stages(ior1), _ = 1:max_scenarios(ior1), _ = 1:max_blocks(ior1)
```
Comment (joaquimg) [inline src/time_series_utils.jl:217]: we probably need to use the version of max blocks that depends on the current stage

File: src/time_series_utils.jl [hunk 2]
```diff
@@ -187,4 +187,38 @@ function file_to_array_and_header(
...
+end
+
+function is_equal(
+    ior1::T1,
+    ior2::T2;
+    atol::Real = 1e-5,
+    rtol::Real = 1e-4
+) where {T1 <: AbstractReader, T2 <: AbstractReader}
+    # Assert have the same stage_type and is_hourly
+    @assert is_hourly(ior1) == is_hourly(ior2) "erro"
```
Comment (joaquimg) [inline src/time_series_utils.jl:199]: More info on the error?

**Review by joaquimg — COMMENTED** [review]
Yep,
Two options I see are: 
1 - to return on the very first error saying where the diff is.
2 - return two values a boolean saying if there is any diff and a string. (the string would contain a list of all the difs in the headers (for the case of name diffs, just return the firs M diffs), and, for instance, the first N diff in the data

File: src/time_series_utils.jl (hunk 1 above)
Comment (guilhermebodin) [reply src/time_series_utils.jl:217]: curiously max_blocks_stage(ior1, estagio) errores

**Review by joaquimg — APPROVED** [review]

**Pull request merged**